In [1]:
import pandas as pd
import numpy as np

import plotly.express as px
import cufflinks as cf
import chart_studio.plotly as py
%matplotlib inline

from plotly.offline import download_plotlyjs, init_notebook_mode,plot,iplot
init_notebook_mode(connected=True)
#to make everything locally
cf.go_offline()

In [2]:
#loading the dataset
data = pd.read_csv("Netflix Dataset.csv")
data.head()

Show_Id Category  Title           Director  \
0      s1  TV Show     3%                NaN   
1      s2    Movie  07:19  Jorge Michel Grau   
2      s3    Movie  23:59       Gilbert Chan   
3      s4    Movie      9        Shane Acker   
4      s5    Movie     21     Robert Luketic   

                                                Cast        Country  \
0  João Miguel, Bianca Comparato, Michel Gomes, R...         Brazil   
1  Demián Bichir, Héctor Bonilla, Oscar Serrano, ...         Mexico   
2  Tedd Chan, Stella Chung, Henley Hii, Lawrence ...      Singapore   
3  Elijah Wood, John C. Reilly, Jennifer Connelly...  United States   
4  Jim Sturgess, Kevin Spacey, Kate Bosworth, Aar...  United States   

        Release_Date Rating   Duration  \
0    August 14, 2020  TV-MA  4 Seasons   
1  December 23, 2016  TV-MA     93 min   
2  December 20, 2018      R     78 min   
3  November 16, 2017  PG-13     80 min   
4    January 1, 2020  PG-13    123 min   

                                                Type  \
0  International TV Shows, TV Dramas, TV Sci-Fi &...   
1                       Dramas, International Movies   
2                Horror Movies, International Movies   
3  Action & Adventure, Independent Movies, Sci-Fi...   
4                                             Dramas   

                                         Description  
0  In a future where the elite inhabit an island ...  
1  After a devastating earthquake hits Mexico Cit...  
2  When an army recruit is found dead, his fellow...  
3  In a postapocalyptic world, rag-doll robots hi...  
4  A brilliant group of students become card-coun...

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7789 entries, 0 to 7788
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Show_Id       7789 non-null   object
 1   Category      7789 non-null   object
 2   Title         7789 non-null   object
 3   Director      5401 non-null   object
 4   Cast          7071 non-null   object
 5   Country       7282 non-null   object
 6   Release_Date  7779 non-null   object
 7   Rating        7782 non-null   object
 8   Duration      7789 non-null   object
 9   Type          7789 non-null   object
 10  Description   7789 non-null   object
dtypes: object(11)
memory usage: 669.5+ KB


In [4]:
data.isnull().sum()

Show_Id            0
Category           0
Title              0
Director        2388
Cast             718
Country          507
Release_Date      10
Rating             7
Duration           0
Type               0
Description        0
dtype: int64

# Missing value Treatment
- there could be some relation between Director and Country, Director and Category, 
- H0: There is no relation between Director and Country
- H1: There is some relation between Director and Country

- H0: There is no relation between Director and Category
- H1: There is some relation between Director and Category

In [5]:
data.Director.value_counts()

Raúl Campos, Jan Suter                                                                                                                                                                 18
Marcus Raboy                                                                                                                                                                           16
Jay Karas                                                                                                                                                                              14
Cathy Garcia-Molina                                                                                                                                                                    13
Youssef Chahine                                                                                                                                                                        12
                                                                      

In [6]:
data.Country.value_counts()

United States                                                                          2556
India                                                                                   923
United Kingdom                                                                          397
Japan                                                                                   226
South Korea                                                                             183
                                                                                       ... 
Ireland, France, Iceland, United States, Mexico, Belgium, United Kingdom, Hong Kong       1
United States, Malta, France, United Kingdom                                              1
Canada, United Kingdom, Netherlands                                                       1
Netherlands, Germany, Italy, Canada                                                       1
United States, Greece, United Kingdom                                           

# Task.1. Is there any Duplicate Record in this dataset ? If yes, then remove the duplicate records.

In [7]:
data[data.duplicated()==True]

Show_Id Category              Title        Director  \
6300    s684    Movie           Backfire     Dave Patten   
6622   s6621    Movie  The Lost Okoroshi  Abba T. Makama   

                                                   Cast        Country  \
6300  Black Deniro, Byron "Squally" Vinson, Dominic ...  United States   
6622  Seun Ajayi, Judith Audu, Tope Tedela, Ifu Enna...        Nigeria   

           Release_Date Rating Duration  \
6300      April 5, 2019  TV-MA   97 min   
6622  September 4, 2020  TV-MA   94 min   

                                       Type  \
6300  Dramas, Independent Movies, Thrillers   
6622   Comedies, Dramas, Independent Movies   

                                            Description  
6300  When two would-be robbers accidentally kill a ...  
6622  A disillusioned security guard transforms into...

In [8]:
data.drop(data[data.duplicated()==True].index , inplace =True)

# Task.2. Is there any Null Value present in any column ? Show with Heat-map.

In [9]:
data.isnull().sum()

Show_Id            0
Category           0
Title              0
Director        2388
Cast             718
Country          507
Release_Date      10
Rating             7
Duration           0
Type               0
Description        0
dtype: int64

In [10]:
px.imshow(data.isnull(),title = "Missing Values" )

# Q.1. For 'House of Cards', what is the Show Id and Who is the Director of this show ?

In [11]:
data.loc[data.Title=='House of Cards']

Show_Id Category           Title  \
2832   s2833  TV Show  House of Cards   

                                               Director  \
2832  Robin Wright, David Fincher, Gerald McRaney, J...   

                                                   Cast        Country  \
2832  Kevin Spacey, Robin Wright, Kate Mara, Corey S...  United States   

          Release_Date Rating   Duration                     Type  \
2832  November 2, 2018  TV-MA  6 Seasons  TV Dramas, TV Thrillers   

                                            Description  
2832  A ruthless politician will stop at nothing to ...

In [12]:
data.loc[data.Title=='House of Cards',['Title', 'Show_Id','Director']]

Title Show_Id  \
2832  House of Cards   s2833   

                                               Director  
2832  Robin Wright, David Fincher, Gerald McRaney, J...

# Q.2. In which year highest number of the TV Shows & Movies were released ? Show with Bar Graph.

In [13]:
data.head(1)

Show_Id Category Title Director  \
0      s1  TV Show    3%      NaN   

                                                Cast Country     Release_Date  \
0  João Miguel, Bianca Comparato, Michel Gomes, R...  Brazil  August 14, 2020   

  Rating   Duration                                               Type  \
0  TV-MA  4 Seasons  International TV Shows, TV Dramas, TV Sci-Fi &...   

                                         Description  
0  In a future where the elite inhabit an island ...

In [14]:
data['Release_year'] = pd.to_datetime(data['Release_Date']).dt.year

In [15]:
data['Release_year'].value_counts().sort_values(ascending = False).values

array([2153, 2009, 1685, 1225,  443,  117,   88,   25,   13,   11,    3,
          2,    2,    1], dtype=int64)

In [16]:
fig = px.bar(data_frame=data, x = data['Release_year'].value_counts().index, y = data['Release_year'].value_counts().values, color =data['Release_year'].value_counts().index ,title = "Number of shows Yearwise")
fig

Answer:
    Year ===== 2019 
    total TV shows+Movies ======== 2153

# Q.3. How many Movies & TV Shows are in the dataset ? Show with Bar Graph.

In [17]:
data.head(1)

Show_Id Category Title Director  \
0      s1  TV Show    3%      NaN   

                                                Cast Country     Release_Date  \
0  João Miguel, Bianca Comparato, Michel Gomes, R...  Brazil  August 14, 2020   

  Rating   Duration                                               Type  \
0  TV-MA  4 Seasons  International TV Shows, TV Dramas, TV Sci-Fi &...   

                                         Description  Release_year  
0  In a future where the elite inhabit an island ...        2020.0

In [18]:
data.groupby(['Category'])['Show_Id'].count().values

array([5377, 2410], dtype=int64)

In [19]:
px.bar(data_frame=data , x =data.groupby(['Category'])['Show_Id'].count().index, 
       y =  data.groupby(['Category'])['Show_Id'].count().values, 
       color = data.groupby(['Category'])['Show_Id'].count().index,
       title = "Movies and TVShows In the Dataset",
       text = data.groupby(['Category'])['Show_Id'].count().values, 
      )

# Q.4. Show all the Movies that were released in year 2000.

In [20]:
data.head(1)

Show_Id Category Title Director  \
0      s1  TV Show    3%      NaN   

                                                Cast Country     Release_Date  \
0  João Miguel, Bianca Comparato, Michel Gomes, R...  Brazil  August 14, 2020   

  Rating   Duration                                               Type  \
0  TV-MA  4 Seasons  International TV Shows, TV Dramas, TV Sci-Fi &...   

                                         Description  Release_year  
0  In a future where the elite inhabit an island ...        2020.0

In [21]:
data.loc[(data.Release_year==2020) & (data.Category=='Movie'),['Category','Title','Release_year']]

Category                                    Title  Release_year
4       Movie                                       21        2020.0
6       Movie                                      122        2020.0
14      Movie                                     3022        2020.0
27      Movie                                   #Alive        2020.0
28      Movie            #AnneFrank - Parallel Stories        2020.0
...       ...                                      ...           ...
7762    Movie                                Zaki Chan        2020.0
7783    Movie                                     Zoom        2020.0
7784    Movie                                     Zozo        2020.0
7786    Movie                        Zulu Man in Japan        2020.0
7788    Movie  ZZ TOP: THAT LITTLE OL' BAND FROM TEXAS        2020.0

[1312 rows x 3 columns]

# Q.5. Show only the Titles of all TV Shows that were released in India only.

In [22]:
data.head(1)

Show_Id Category Title Director  \
0      s1  TV Show    3%      NaN   

                                                Cast Country     Release_Date  \
0  João Miguel, Bianca Comparato, Michel Gomes, R...  Brazil  August 14, 2020   

  Rating   Duration                                               Type  \
0  TV-MA  4 Seasons  International TV Shows, TV Dramas, TV Sci-Fi &...   

                                         Description  Release_year  
0  In a future where the elite inhabit an island ...        2020.0

In [23]:
data.loc[(data['Country']=='India') & (data['Category']=='TV Show'),['Title']]

Title
86          21 Sarfarosh: Saragarhi 1897
132                            7 (Seven)
340                         Agent Raghav
364                         Akbar Birbal
533                  Anjaan: Rural Myths
...                                  ...
6249                The Creative Indians
6400  The Golden Years with Javed Akhtar
6469              The House That Made Me
7294                          Typewriter
7705                     Yeh Meri Family

[71 rows x 1 columns]

# Show Top 10 Directors, who gave the highest number of TV Shows & Movies to Netflix ?

In [24]:
data.Director.value_counts().sort_values(ascending = False).index[:10]

Index(['Raúl Campos, Jan Suter', 'Marcus Raboy', 'Jay Karas',
       'Cathy Garcia-Molina', 'Martin Scorsese', 'Jay Chapman',
       'Youssef Chahine', 'Steven Spielberg', 'David Dhawan', 'Hakan Algül'],
      dtype='object')

In [25]:
px.bar(data, x = data.Director.value_counts().sort_values(ascending = False).index[:10], 
        y= data.Director.value_counts().sort_values(ascending = False).values[:10], 
        color =data.Director.value_counts().sort_values(ascending = False).values[:10], 
        title = 'Top 10 Directors with Highest Number of TV Shows & Movies' )

# Q.7. Show all the Records, where "Category is Movie and Type is Comedies" or "Country is United Kingdom".

In [26]:
data.head(1)

Show_Id Category Title Director  \
0      s1  TV Show    3%      NaN   

                                                Cast Country     Release_Date  \
0  João Miguel, Bianca Comparato, Michel Gomes, R...  Brazil  August 14, 2020   

  Rating   Duration                                               Type  \
0  TV-MA  4 Seasons  International TV Shows, TV Dramas, TV Sci-Fi &...   

                                         Description  Release_year  
0  In a future where the elite inhabit an island ...        2020.0

In [27]:
data[(data.Type=='Comedies')&(data.Category=='Movie') | (data.Country=='United Kingdom')]

Show_Id Category                   Title          Director  \
19       s20    Movie                     '89               NaN   
33       s34    Movie            #realityhigh  Fernando Lebrija   
58       s59  TV Show             100% Hotter               NaN   
72       s73    Movie                17 Again       Burr Steers   
82       s83    Movie     2036 Origin Unknown     Hasraf Dulull   
...      ...      ...                     ...               ...   
7670   s7669  TV Show  World War II in Colour               NaN   
7671   s7670  TV Show  World's Busiest Cities               NaN   
7688   s7687    Movie  XV: Beyond the Tryline  Pierre Deschamps   
7725   s7724    Movie            You Can Tutu       James Brown   
7740   s7739  TV Show         Young Wallander               NaN   

                                                   Cast         Country  \
19                   Lee Dixon, Ian Wright, Paul Merson  United Kingdom   
33    Nesta Cooper, Kate Walsh, John Michael Higgins...   United States   
58    Daniel Palmer, Melissa Sophia, Karen Williams,...  United Kingdom   
72    Zac Efron, Leslie Mann, Matthew Perry, Thomas ...   United States   
82    Katee Sackhoff, Ray Fearon, Julie Cox, Steven ...  United Kingdom   
...                                                 ...             ...   
7670                                      Robert Powell  United Kingdom   
7671                 Anita Rani, Ade Adepitan, Dan Snow  United Kingdom   
7688                                                NaN  United Kingdom   
7725  Lily O'Regan, Jeannettsy Enriquez Borges, Joel...  United Kingdom   
7740  Adam Pålsson, Richard Dillane, Leanne Best, El...  United Kingdom   

           Release_Date Rating  Duration  \
19         May 16, 2018  TV-PG    87 min   
33    September 8, 2017  TV-14    99 min   
58     November 1, 2019  TV-14  1 Season   
72      January 1, 2021  PG-13   102 min   
82    December 20, 2018  TV-14    95 min   
...                 ...    ...       ...   
7670     August 1, 2017  TV-MA  1 Season   
7671   February 1, 2019  TV-PG  1 Season   
7688     March 18, 2020  TV-14    91 min   
7725  December 31, 2017   TV-G    87 min   
7740  September 3, 2020  TV-MA  1 Season   

                                                   Type  \
19                                        Sports Movies   
33                                             Comedies   
58    British TV Shows, International TV Shows, Real...   
72                                             Comedies   
82                                     Sci-Fi & Fantasy   
...                                                 ...   
7670  British TV Shows, Docuseries, International TV...   
7671                       British TV Shows, Docuseries   
7688                       Documentaries, Sports Movies   
7725                           Children & Family Movies   
7740  Crime TV Shows, International TV Shows, TV Dramas   

                                            Description  Release_year  
19    Mixing old footage with interviews, this is th...        2018.0  
33    When nerdy high schooler Dani finally attracts...        2017.0  
58    A stylist, a hair designer and a makeup artist...        2019.0  
72    Nearing a midlife crisis, thirty-something Mik...        2021.0  
82    Working with an artificial intelligence to inv...        2018.0  
...                                                 ...           ...  
7670  Footage of the most dramatic moments from Worl...        2017.0  
7671  From Moscow to Mexico City, three BBC journali...        2019.0  
7688  Set against the 2015 Rugby World Cup, this doc...        2020.0  
7725  A gifted young ballet dancer struggles to find...        2017.0  
7740  An incendiary hate crime stirs civil unrest, f...        2020.0  

[485 rows x 12 columns]

# Q.8. In how many movies/shows, Tom Cruise was cast ?

In [28]:
data.head(1)

Show_Id Category Title Director  \
0      s1  TV Show    3%      NaN   

                                                Cast Country     Release_Date  \
0  João Miguel, Bianca Comparato, Michel Gomes, R...  Brazil  August 14, 2020   

  Rating   Duration                                               Type  \
0  TV-MA  4 Seasons  International TV Shows, TV Dramas, TV Sci-Fi &...   

                                         Description  Release_year  
0  In a future where the elite inhabit an island ...        2020.0

In [29]:
data[data.Cast.str.contains('Tom Cruise').fillna(False)]

Show_Id Category     Title              Director  \
3860   s3861    Movie  Magnolia  Paul Thomas Anderson   
5071   s5071    Movie  Rain Man        Barry Levinson   

                                                   Cast        Country  \
3860  Jeremy Blackman, Tom Cruise, Melinda Dillon, A...  United States   
5071  Dustin Hoffman, Tom Cruise, Valeria Golino, Ge...  United States   

         Release_Date Rating Duration                        Type  \
3860  January 1, 2020      R  189 min  Dramas, Independent Movies   
5071     July 1, 2019      R  134 min      Classic Movies, Dramas   

                                            Description  Release_year  
3860  Through chance, human action, past history and...        2020.0  
5071  A fast-talking yuppie is forced to slow down w...        2019.0

# Q.9. What are the different Ratings defined by Netflix ?

In [30]:
print("Total number of different ratings are " ,data['Rating'].nunique() ,"\nThey are ",data['Rating'].unique())

Total number of different ratings are  14 
They are  ['TV-MA' 'R' 'PG-13' 'TV-14' 'TV-PG' 'NR' 'TV-G' 'TV-Y' nan 'TV-Y7' 'PG'
 'G' 'NC-17' 'TV-Y7-FV' 'UR']


# Q.9.1. How many Movies got the 'TV-14' rating, in Canada ?

In [31]:
data.loc[(data.Rating=='TV-14') & (data.Country=='Canada') &  (data.Category=='Movie'), 'Category'].count()

11

# Q.9.2. How many TV Show got the 'R' rating, after year 2018 ?

In [32]:
data.head(1)

Show_Id Category Title Director  \
0      s1  TV Show    3%      NaN   

                                                Cast Country     Release_Date  \
0  João Miguel, Bianca Comparato, Michel Gomes, R...  Brazil  August 14, 2020   

  Rating   Duration                                               Type  \
0  TV-MA  4 Seasons  International TV Shows, TV Dramas, TV Sci-Fi &...   

                                         Description  Release_year  
0  In a future where the elite inhabit an island ...        2020.0

In [33]:
data.loc[(data.Rating=='R') & (data.Category=='TV Show') & (data.Release_year > 2018),'Show_Id'].count()

1

# Q.10. What is the maximum duration of a Movie/Show on Netflix ?

In [34]:
#Creating a series  and stories all duration which is availale in minutes 
minutes = pd.Series(data.loc[data.Duration.str.contains('min'), 'Duration'])
#Creating a series  and stories all duration which is availale in seasons 
seasons = pd.Series(data.loc[data.Duration.str.contains('Seasons'), 'Duration'])

In [35]:
def clean(series):
    cleanlist = []
    for i in series.unique():
        cleanlist.append(int(list(i.split(" "))[0]))
    cleanlist.sort(reverse = True)
    return cleanlist

In [36]:
maximum_minutes = clean(minutes)[0]
maximum_seasons = clean(seasons)[0]
print("Movie/Show with maximum minutes", maximum_minutes, "\n Movie/Show with maximum Seasons", maximum_seasons)

Movie/Show with maximum minutes 312 
 Movie/Show with maximum Seasons 16


# Q.11. Which individual country has the Highest No. of TV Shows ?

In [37]:
fig=px.bar(data,
       x = data.Country.value_counts().sort_values(ascending = False).index[:20],
       y = data.Country.value_counts().sort_values(ascending = False).values[:20],
    color = data.Country.value_counts().sort_values(ascending = False).index[:20],
        title  = "Country Wise Total Shows",
          text = data.Country.value_counts().sort_values(ascending = False).values[:20])
fig.update_traces(textposition = 'outside')

# Q.12. How can we sort the dataset by Year ?

In [38]:
data.sort_values(by = 'Release_year')

Show_Id Category                         Title              Director  \
7115   s7114    Movie          To and From New York  Sorin Dan Mihalcescu   
1765   s1766  TV Show               Dinner for Five                   NaN   
3248   s3249    Movie       Just Another Love Story          Ole Bornedal   
5766   s5766    Movie                      Splatter             Joe Dante   
3840   s3841    Movie   Mad Ron's Prevues from Hell            Jim Monaco   
...      ...      ...                           ...                   ...   
3374   s3375  TV Show                      Kikoriki                   NaN   
3492   s3493  TV Show           La Familia P. Luche                   NaN   
3946   s3947  TV Show                         Maron                   NaN   
5138   s5138  TV Show                  Red vs. Blue                   NaN   
6066   s6066  TV Show  The Adventures of Figaro Pho                   NaN   

                                                   Cast        Country  \
7115  Barbara King, Shaana Diya, John Krisiukenas, Y...  United States   
1765                                                NaN  United States   
3248  Anders W. Berthelsen, Rebecka Hemse, Nikolaj L...        Denmark   
5766  Corey Feldman, Tony Todd, Tara Leigh, Erin Way...  United States   
3840  Nick Pawlow, Jordu Schell, Jay Kushwara, Micha...  United States   
...                                                 ...            ...   
3374                                      Igor Dmitriev            NaN   
3492  Eugenio Derbez, Consuelo Duval, Luis Manuel Áv...  United States   
3946  Marc Maron, Judd Hirsch, Josh Brener, Nora Zeh...  United States   
5138  Burnie Burns, Jason Saldaña, Gustavo Sorola, G...  United States   
6066  Luke Jurevicius, Craig Behenna, Charlotte Haml...      Australia   

           Release_Date Rating    Duration  \
7115    January 1, 2008  TV-MA      81 min   
1765   February 4, 2008  TV-MA    1 Season   
3248        May 5, 2009  TV-MA     104 min   
5766  November 18, 2009  TV-MA      29 min   
3840   November 1, 2010     NR      84 min   
...                 ...    ...         ...   
3374                NaN   TV-Y   2 Seasons   
3492                NaN  TV-14   3 Seasons   
3946                NaN  TV-MA   4 Seasons   
5138                NaN     NR  13 Seasons   
6066                NaN  TV-Y7   2 Seasons   

                                                   Type  \
7115              Dramas, Independent Movies, Thrillers   
1765                       Stand-Up Comedy & Talk Shows   
3248                       Dramas, International Movies   
5766                                      Horror Movies   
3840                         Cult Movies, Horror Movies   
...                                                 ...   
3374                                           Kids' TV   
3492  International TV Shows, Spanish-Language TV Sh...   
3946                                        TV Comedies   
5138  TV Action & Adventure, TV Comedies, TV Sci-Fi ...   
6066                              Kids' TV, TV Comedies   

                                            Description  Release_year  
7115  While covering a story in New York City, a Sea...        2008.0  
1765  In each episode, four celebrities join host Jo...        2008.0  
3248  When he causes a car accident that leaves a yo...        2009.0  
5766  After committing suicide, a washed-up rocker r...        2009.0  
3840  This collection cherry-picks trailers, forgott...        2010.0  
...                                                 ...           ...  
3374  A wacky rabbit and his gang of animal pals hav...           NaN  
3492  This irreverent sitcom featues Ludovico, Feder...           NaN  
3946  Marc Maron stars as Marc Maron, who interviews...           NaN  
5138  This parody of first-person shooter games, mil...           NaN  
6066  Imagine your worst fears, then multiply them: ...           NaN  

[7787 rows x 12 columns]

# Q.13. Find all the instances where : Category is 'Movie' and Type is 'Dramas' or Category is 'TV Show' & Type is 'Kids' TV'.

In [39]:

data[((data.Category=='Movie') & (data.Type=='Dramas') )| ((data.Category=='TV Show') & (data.Type.str.contains('Kids')) )]

Show_Id Category                     Title        Director  \
4         s5    Movie                        21  Robert Luketic   
7         s8    Movie                       187  Kevin Reynolds   
108     s109  TV Show  3Below: Tales of Arcadia             NaN   
111     s112  TV Show                   44 Cats             NaN   
170     s171    Movie              A Family Man   Mark Williams   
...      ...      ...                       ...             ...   
7758   s7757  TV Show                        Z4             NaN   
7761   s7760  TV Show                 Zak Storm             NaN   
7768   s7767  TV Show              Zig & Sharko             NaN   
7777   s7776  TV Show                Zoids Wild             NaN   
7779   s7778  TV Show               Zombie Dumb             NaN   

                                                   Cast  \
4     Jim Sturgess, Kevin Spacey, Kate Bosworth, Aar...   
7     Samuel L. Jackson, John Heard, Kelly Rowan, Cl...   
108   Tatiana Maslany, Diego Luna, Nick Offerman, Ni...   
111   Sarah Natochenny, Suzy Myers, Simona Berman, E...   
170   Gerard Butler, Gretchen Mol, Alison Brie, Will...   
...                                                 ...   
7758  Apollo Costa, Gabriel Santana, Matheus Lustosa...   
7761  Michael Johnston, Jessica Gee-George, Christin...   
7768                                                NaN   
7777  Kensho Ono, Takahiro Sakurai, Mikako Komatsu, ...   
7779                                                NaN   

                                            Country        Release_Date  \
4                                     United States     January 1, 2020   
7                                     United States    November 1, 2019   
108                                   United States       July 12, 2019   
111                                           Italy     October 1, 2020   
170                           Canada, United States   December 15, 2019   
...                                             ...                 ...   
7758                                         Brazil   February 22, 2019   
7761  United States, France, South Korea, Indonesia  September 13, 2018   
7768                                         France    December 1, 2017   
7777                                          Japan     August 14, 2020   
7779                                            NaN        July 1, 2019   

     Rating   Duration                                               Type  \
4     PG-13    123 min                                             Dramas   
7         R    119 min                                             Dramas   
108   TV-Y7  2 Seasons  Kids' TV, TV Action & Adventure, TV Sci-Fi & F...   
111   TV-Y7  2 Seasons                                           Kids' TV   
170       R    110 min                                             Dramas   
...     ...        ...                                                ...   
7758  TV-PG  2 Seasons                                           Kids' TV   
7761  TV-Y7  3 Seasons                                           Kids' TV   
7768  TV-Y7   1 Season                              Kids' TV, TV Comedies   
7777  TV-Y7   1 Season                             Anime Series, Kids' TV   
7779  TV-Y7  2 Seasons             Kids' TV, Korean TV Shows, TV Comedies   

                                            Description  Release_year  
4     A brilliant group of students become card-coun...        2020.0  
7     After one of his high school students attacks ...        2019.0  
108   After crash-landing on Earth, two royal teen a...        2019.0  
111   Paw-esome tales abound when singing furry frie...        2020.0  
170   A ruthless corporate headhunter battles his ri...        2019.0  
...                                                 ...           ...  
7758  Fading music biz veteran Zé realizes he has ju...        2019.0  
7761  Teen surfer Zak Storm is mysteriously transpor...        2018.0  
7768  Zig, an isl